Projects/b_u585i_iot02a_ntz/Src/ota_pal/ota_pal_stm32u5_ntz.c

```diff
@@ -375,390 +591,570 @@ static HAL_StatusTypeDef prvWriteToFlash( uint32_t destination,
+                if( ( pxContext->xPalState == OTA_PAL_PENDING_SELF_TEST ) ||
+                    ( pxContext->xPalState == OTA_PAL_NEW_IMAGE_BOOTED ) )
                 {
-                    /** New image bank is booted successfully and it have passed self test. Make it as accepted
-                     * by removing the image context from flash and setting image state to valid. */
-
-                    prvRemoveImageContextFromFlash();
-
-                    ( void ) prvEraseBank( passiveBank );
-                    pContext->state = OTA_PAL_IMAGE_STATE_COMMITTED;
-                    otaStatus = OtaPalSuccess;
+                    xOtaStatus = OtaPalSuccess;
+                    pxContext->xPalState = OTA_PAL_ACCEPTED;
+
+                    /* Delete context from flash */
+                    if( prvDeletePalNvContext() != pdTRUE )
+                    {
+                        xOtaStatus = OtaPalCommitFailed;
+                    }
                 }
                 else
                 {
-                    otaStatus = OtaPalCommitFailed;
+                    xOtaStatus = OtaPalCommitFailed;
                 }
 
                 break;
 
             case OtaImageStateRejected:
 
-                /* Make sure the image is not committed already. A committed image cannot be aborted or rejected. */
-                if( pContext->state != OTA_PAL_IMAGE_STATE_COMMITTED )
+                switch( pxContext->xPalState )
                 {
-                    /*
-                     *  Remove the persisted state of image from flash.
-                     */
-                    prvRemoveImageContextFromFlash();
-                    pContext->state = OTA_PAL_IMAGE_STATE_INVALID;
-                    otaStatus = OtaPalSuccess;
+                    case OTA_PAL_PENDING_SELF_TEST:
+                    case OTA_PAL_NEW_IMAGE_BOOTED:
+                    case OTA_PAL_NEW_IMAGE_WDT_RESET:
+                    case OTA_PAL_PENDING_ACTIVATION:
+                    case OTA_PAL_FILE_OPEN:
+
+                        if( ( prvSelectBank( ulGetOtherBank( pxContext->ulTargetBank ) ) == pdTRUE ) &&
+                            ( prvEraseBank( pxContext->ulTargetBank ) == pdTRUE ) &&
+                            ( prvDeletePalNvContext() == pdTRUE ) )
+                        {
+                            xOtaStatus = OtaPalSuccess;
+                            pxContext->xPalState = OTA_PAL_REJECTED;
+                        }
+
+                        break;
+
+                    case OTA_PAL_REJECTED:
+                        xOtaStatus = OtaPalSuccess;
+                        break;
+
+                    default:
+                        break;
                 }
-                else
+
+                break;
+
+            case OtaImageStateTesting:
+
+                switch( pxContext->xPalState )
                 {
-                    otaStatus = OtaPalRejectFailed;
+                    case OTA_PAL_PENDING_ACTIVATION:
```

Comment:
The state is set after the new image is booted is booted up and OTA agent gets a self test job document.  So the state should probably be never hit in this case.